VERSION

```diff
@@ -0,0 +1 @@
+8.1.0
```

Comment:
How is this VERSION file used?  Does cmake or workflows queue off of this or...?  Each application/model implementation uses a different release version number, so I'm wondering if it makes sense to have this file in develop and if yes, what that version number should be so as not to confuse other applications.  For example, in public/release-v1 it was set to 1.0.0.   What does 8.1.0 refer to?

Reply:
Version on the develop branch is `<last-release>+dev`.
I guessed `8.1.0` from Hera (`8.0.6`)
I suppose one can add `+dev` to whatever the `last-release` is.

Reply:
Ok, so post library version, presumably?  But when making a branch for another release application, this file will change to the release version number (e.g. 1.0.0 for release/public-v1), which may be confusing. Is this VERSION file necessary to exist in develop? What uses this file? Or is it for informational purposes?  You can see some extra UPP release version info on the wiki: https://github.com/NOAA-EMC/EMC_post/wiki/UPP-Release

Reply:
@fossell It is the package version, whatever you call a package -- library or application.  I don't think you can have a different version of a library and an application reside in the same package, or atleast I cannot think of an example.
  
A version **must** be provided with the package.  That is just standard practice.
There are different ways to do this; I can think of 2.
- Like the one here, provide an external file and maintain it.
- Hard-wire the version number in CMakeLists.txt and update it (Unidata does this in netcdf-c)

The issue with the latter is, when issuing different releases/tags, you may rarely need to update the top-level `CMakeLists.txt`, unless you add or remove a dependency or directory, etc.  Most changes are confined to lower levels where the source code changes.  So you will have to make a change in this file, just for the sake of incrementing the version number. And this will show up as a change in the git log.

In the end, it is just a preference, how one wishes to maintain the versioning of the software.  I prefer the former.

Reply:
Thanks for the info.  Unfortunately the version/release procedures for UPP are not as straight forward. This was discussed in detail during UFS V1.0 release, but perhaps @WenMeng-NOAA can provide a bit of background on the impacts for implementation of operational models. I'm just not sure what we would put in VERSION for develop branch that would be accurate and consistent for all apps.